boa/src/builtins/function/mod.rs

```diff
@@ -329,6 +329,79 @@ impl BuiltInFunctionObject {
         // TODO?: 5. PrepareForTailCall
         context.call(this, &this_arg, &arg_list)
     }
+
+    #[allow(clippy::wrong_self_convention)]
+    fn to_string(this: &Value, _: &[Value], context: &mut Context) -> Result<Value> {
+        let name = {
+            // Is there a case here where if there is no name field on a value
+            // We it should default to None? Do all functions have names set?
```

Comment:
Minor suggestion on phrasing.
```suggestion
            // Is there a case here where if there is no name field on a value
            // name should default to None? Do all functions have names set?
```
And to answer, no, not all functions have names, certain arrow functions don't have names.
As an example, with:
```js
let d = (() => {
  return () => {
    return 3.14;
  };
})
```
You get
```js
d().toString() == "() => {
  return 3.14;
}"
```

And `d().name == ""`